wi

```diff
@@ -54,17 +86,32 @@ async def download(url, dest):
             open(dest, 'wb+').write(await response.read())
             log('Downloaded ' + url)
 
-async def install(name, version):
+
+async def install(repo_url, name, version):
     """Install a wheel package into the system. It will try to use the
     best match for the machine, considering the architecture and OS.
 
-    If the package can't be found, 
+    If the package can't be found,
     """
-    package = await metadata(name, version)
-    if not package:
-        return ('missing', name, version)
-    releases = package['releases'][package['info']['version']]
-    with tempfile.TemporaryDirectory() as tmpdir: 
+    package = await metadata(repo_url, name, version)
+    parser = PyPIParser()
+    parser.feed(package)
+    # if not package:
+    #     return ('missing', name, version)
+    # releases = package['releases'][package['info']['version']]
+    releases = parser.all_data
+    # Create the full URLs
+    for release in releases:
+        if 'http' in release['href']:
+            release['url'] = release['href']
+        else:
+            release['url'] = '/'.join([repo_url, release['href']])
+
+    # Filter by version, if required
+    if version:
+        releases = [r for r in releases if version in r['filename']]
+
```

Comment:
Perhaps:
```
if version:
    releases = filter(lambda release: release['filename'].rstrip('-', 1)[1].startswith(version),
        filter(lambda release: '-' in release, releases))
```

Reply:
I've changed it to a more explicit:

```
# The format of the wheel filename is, (PEP 427)
# {distribution}-{version}(-{build tag})?-{python tag}-
# {abi tag}-{platform tag}.whl
WHEEL_REGEX = r'(?P<distribution>[\w\d.]+)-(?P<version>[\d.]+)-(\S+).whl'
WHEEL_MATCH = re.compile(WHEEL_REGEX)

def check_version_in_filename(filename, version):
    """
    Checks that the wheel filename has the proper version
    """
    result = re.match(WHEEL_MATCH, filename)
    if not result:
        # Not a wheel file
        return False

    return result.groupdict()['version'] == version
...
    if version:
        releases = [r for r in releases
                    if check_version_in_filename(r['filename'], version)]
```

(I know about regexes, but I think in this case it makes an explicit parsing of the filename

Reply:
Another option is going for the more obvious
```
version = filename.split('-')[1]
```
But it looks more "magical" to me, but maybe I'm overthinking it...